cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -371,6 +454,9 @@ private void restartIfNecessary(NodeRef nodeRef, RestartContext restartContext)
                 await(isReady(pod), operationTimeoutMs, TimeUnit.MILLISECONDS, e -> new RuntimeException(e));
             } catch (Exception e)   {
                 if (e.getCause() instanceof TimeoutException) {
+                    if (getBrokerState(pod.getMetadata().getName()) == (BROKER_RECOVERY_STATE)) {
+                        throw new UnforceableProblem("Pod " + nodeRef.podName() + " is not ready because the broker is performing log recovery.", e.getCause());
+                    }
```

Comment:
I wonder if we should try the check for every exception and not just timeout ...